code-studio/src/js/components/progress/progress_dot.jsx

```diff
@@ -138,12 +138,40 @@ const styles = {
   }
 };
 
-
 function dotClicked(url, e) {
   e.preventDefault();
   saveAnswersAndNavigate(url);
 }
 
+const BubbleInterior = React.createClass({
```

Comment:
Yeah, I think one-class-per-file should be the general rule. I think to qualify for an exception the following needs to hold true.
1. Component is used in only this file
2. Component has a reasonably simple render method, and no other methods.